### PR TITLE
Testing environment fixes

### PR DESCRIPTION
**Pull request by vhavlena**
This PR makes a couple of changes in the testing environment of mata:

- fixing examples to be compatible with the current version of mata
- extension of the environment for the correctness checking
- some minor structural changes

A question to consider is whether we want to include the correctness tests on a selected subset of automata to our CI at some point? I mean, so far it is fine, but later if someone starts using mata, we should minimize the probability of introducing bugs in the main branch.

### CODE REVIEW

**Comment by Adda0** [comment]
I would include the correctness tests in our CI. It costs us nothing, yet it may save us time in the future.

I have fixed examples in one of my PRs as well. The PR which will be merged second will have to resolve merge conflicts.

**Review by Adda0 — APPROVED** [review]
I like the new infrastructure for correctness testing. We may want to extract some of the duplicated code in performance test source files, but that is a concern of future PRs.

File: tests-performance/src/test-unary-op.cc [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    Nfa aut;
+    Mata::StringToSymbolMap stsm;
+    if (load_automaton(filename, aut, stsm) != EXIT_SUCCESS) {
+        return EXIT_FAILURE;
+    }
+
+    // trim test
+    Nfa aut_trim(aut);
+    aut_trim.trim();
+    std::cout << "trim:" << (Mata::Nfa::are_equivalent(aut, aut_trim) ? "ok" : "fail") << std::endl;
```
Comment (tfiedor) [inline tests-performance/src/test-unary-op.cc:80]: Do we want to only test the correctness, or measure the speed as well? If latter, I would suggest to wrap the code with measurement (you can copy the lines from other templates).

**Review by tfiedor — APPROVED** [review]
Sorry, for late review, I am OK with the PR, I like the proposed changes.

1. I do not require to include this in the CI/CD as suggested by @Adda0. While I think it should be included in the CI/CD, after my vacation I will be making some changes to build enviroment anyway, and in the process I will integrate this to CI/CD as well.
2. I am wondering if it would be usefull to have some macro (similar to catch) that would easily measure some parts of code. What do you think? Would you find it useful as well?
E.g.
```
TIMEIT("minimization", aut_min = Mata::Nfa::minimize(aut_min));
```
I do not require this in this PR, just raising question for some future PR.

File: tests-performance/src/test-unary-op.cc (hunk 1 above)
Comment (vhavlena) [reply tests-performance/src/test-unary-op.cc:80]: Yes, the information might be interesting. I add it. But in the CI/CD I would employ only this binary output. It is difficult to measure some degradation, no? You made a business of it.

**Comment by vhavlena** [comment]
> Sorry, for late review, I am OK with the PR, I like the proposed changes.
> 
> 1. I do not require to include this in the CI/CD as suggested by @Adda0. While I think it should be included in the CI/CD, after my vacation I will be making some changes to build enviroment anyway, and in the process I will integrate this to CI/CD as well.
> 2. I am wondering if it would be usefull to have some macro (similar to catch) that would easily measure some parts of code. What do you think? Would you find it useful as well?
>    E.g.
> 
> ```
> TIMEIT("minimization", aut_min = Mata::Nfa::minimize(aut_min));
> ```
> 
> I do not require this in this PR, just raising question for some future PR.

1. Great! We should also have a discussion on the automata selection later (covering a lot of interesting cases, but not killing github servers). 
2. That could be definitely useful, especially in the performance testing.

**Comment by Adda0** [comment]
Ad 2. We have a similar thing in Z3 for logging, and we find that useful. We could probably make use of said timing macro here, too.

**Review by kilohsakul — APPROVED** [review]
I don't have any comments, I would need more context. Seems good. 

Is this related to selecting benchmarks to the benchmark repo? 

Having some macro for easy measuring would be nice I guess.

**Comment by Adda0** [comment]
@vhavlena Is this PR ready for merge? If so, please let me know, and feel free to go through the ideas suggested in this PR and create issues from the suggestions you find useful for us. We can work on them in the future.

**Comment by vhavlena** [comment]
Yes, I think it is ready for merge. In the following work we should somehow integrate it inside CI/CD (and implement the TIMEIT as @tfiedor suggested).

**Pull request merged**